### PR TITLE
Rewording/changes to bullet points conclusion

### PR DESCRIPTION
**Pull request by ryanford-dev**
Working on the wordage a little. Updated some of the bullet points that might have been off target. Open to further tweaking.

### CODE REVIEW

File: web_development_101/tying_it_all_together/conclusion.md [hunk 1]
```diff
@@ -1,43 +1,44 @@
...
-  - You will learn things "The Ruby Way", which can help you write clean code in other languages as well.
-* Elegant
-  - Brings a lot of tools to help you develop things in a way that makes your code easy to read.
-  - Often times, solutions in Ruby are much more concise than other languages.
+* Convention over configuration
+  - You don't have to make decisions on the structure of your projects.
+  - Others Ruby code tend to be similarly structured and thus easy to navigate.
+* Expressive and flexible syntax
+  - Often code can be written in a human-readable style.
+  - Easy to write DSLs or modify existing classes for your usecase.
```
Comment (I3uckwheat) [inline web_development_101/tying_it_all_together/conclusion.md:21]: I don't think people will know what a DSL is at this point.

**Review by I3uckwheat — CHANGES_REQUESTED** [review]

File: web_development_101/tying_it_all_together/conclusion.md [hunk 2]
```diff
@@ -1,43 +1,44 @@
...
 #### Ruby and Ruby On Rails: 
-* Opinionated 
-  - You don't have to make decisions on the structure of your projects. This helps when moving into projects built by others.
-  - You will learn things "The Ruby Way", which can help you write clean code in other languages as well.
-* Elegant
-  - Brings a lot of tools to help you develop things in a way that makes your code easy to read.
-  - Often times, solutions in Ruby are much more concise than other languages.
+* Convention over configuration
+  - You don't have to make decisions on the structure of your projects.
+  - Others Ruby code tend to be similarly structured and thus easy to navigate.
```
Comment (I3uckwheat) [inline web_development_101/tying_it_all_together/conclusion.md:18]: Others' ?

File: web_development_101/tying_it_all_together/conclusion.md (hunk 1 above)
Comment (ryanford-dev) [reply web_development_101/tying_it_all_together/conclusion.md]: Good call

File: web_development_101/tying_it_all_together/conclusion.md (hunk 2 above)
Comment (ryanford-dev) [reply web_development_101/tying_it_all_together/conclusion.md]: Indeed

**Comment by KevinMulhern** [comment]
Hey @ryanford-dev, sorry this file has been moved to foundations//tying_it_all_together/conclusion.md

**Review by I3uckwheat — APPROVED** [review]

File: foundations/tying_it_all_together/conclusion.md [hunk 3]
```diff
@@ -1,43 +1,44 @@
...
 
-You should be fairly comfortable typing commands into your computer's command line and you should understand what happens when you enter a web address into your browser and it returns a functioning page.  You should be able to create a basic web page and make it do some basic things with JavaScript.  You should understand the purpose of backend code and how frameworks make it easier to write clean and maintainable code.  Finally, you should have a clear understanding of how all this fits together in your future as a builder of web applications.
+You should be fairly comfortable typing commands into your computer's command line and you should understand what happens when you enter a web address into your browser and it returns a page.  You should be able to create a basic web page and make it interactive with JavaScript.  You should understand the purpose of backend code and how frameworks make it easier to bootstrap new projects.  Finally, you should have a clear understanding of how all this fits together in your future as a builder of web applications.
```
Comment (I3uckwheat) [inline foundations/tying_it_all_together/conclusion.md:3]: instead of "interactive with javascript" it should say "interactive using javascript"

A beginner has no idea what "bootstrap new projects" means. I think this should be left as-is.

File: foundations/tying_it_all_together/conclusion.md [hunk 4]
```diff
@@ -1,43 +1,44 @@
...
-  - You will learn things "The Ruby Way", which can help you write clean code in other languages as well.
-* Elegant
-  - Brings a lot of tools to help you develop things in a way that makes your code easy to read.
-  - Often times, solutions in Ruby are much more concise than other languages.
+* Convention over configuration
+  - You don't have to make decisions on the structure of your projects.
+  - Others' Ruby code tend to be similarly structured and thus easy to navigate.
+* Expressive and flexible syntax
+  - Often code can be written in a human-readable style.
+  - Easy to modify existing classes for your usecase.
```
Comment (I3uckwheat) [inline foundations/tying_it_all_together/conclusion.md:21]: I don't think monkey-patching is a good thing

File: foundations/tying_it_all_together/conclusion.md [hunk 5]
```diff
@@ -1,43 +1,44 @@
...
 * Rapid development
-  - Rails helps you build applications very quickly, this means you can make a greater amount of interesting projects in the same time frame as others.
-* Single paradigm
-  - OOP is the Ruby way, many other languages are primarily OOP based.
+  - Rails is "batteries-included". It comes with solutions to common problems already bundled and ready to use.
+  - Rails can generate project boilerplate with it's scaffolding generators and CLI commands before you even open your editor.
```
Comment (I3uckwheat) [inline foundations/tying_it_all_together/conclusion.md:24]: New people will have no idea what this means. We need more beginner friendly terms.

File: foundations/tying_it_all_together/conclusion.md [hunk 6]
```diff
@@ -1,43 +1,44 @@
...
 
 #### JavaScript/Node & Express
 * Un-opinionated
   - There is no standard way of doing things, you are going to have decide structure of projects yourself, for better or worse.
+* Large and Still Growing Community
+  - New, innovative tooling and libraries being developed all the time.
+  - Lots of new employment opportunities using JS.
 * Very fast
-  - Because JavaScript is JIT compiled, it can do calculations very quickly.
-* Verbose
```
Comment (I3uckwheat) [inline foundations/tying_it_all_together/conclusion.md:32]: I think this is important to leave. People should know that JS is much more verbose in it's code.

**Review by I3uckwheat — CHANGES_REQUESTED** [review]

File: foundations/tying_it_all_together/conclusion.md [hunk 7]
```diff
@@ -1,43 +1,44 @@
...
 
-Learning how to program is **much** more than just learning a language. It's about learning how to research and solve difficult problems, building an engineering mindset, working on complex problems and codebases, communicating effectively with other developers, and **much** more. Luckily, all these skills are transferrable to other languages and domains of software and because most languages are similar in concept and structure, moving into a different language will be much easier than learning your first.
+Learning how to program is **much** more than just learning a language. It's about learning how to break down your problems into achievable steps, research and solve difficult problems, communicating effectively with other developers, and much more. Luckily, all these skills are transferrable to other languages and domains of software and because most languages are similar in concept and structure, moving into a different language will be much easier than learning your first.
```
Comment (marvingay) [inline foundations/tying_it_all_together/conclusion.md:9]: I like this, however, it should be an amalgamation of what was previously there and your new edits. We're presenting these on a high level so the concept of having an "engineering mindset" is something that the learner should be made aware of before they move forward.

File: foundations/tying_it_all_together/conclusion.md [hunk 8]
```diff
@@ -1,43 +1,44 @@
...
 
 Arguments about learning the languages used in your area may sound convincing, but because a _good_ developer can transfer skills easily, what you learn isn't going to massively change outcomes of getting a job. Though, if you cannot shake that fear, do whatever you need, even if that includes researching your area to see what jobs are available locally (don't forget about remote jobs though). A good course is far more important than the language you learn first. 
 
 To help you make a decision, here are some perks of each technology:
 
 #### Ruby and Ruby On Rails: 
-* Opinionated 
-  - You don't have to make decisions on the structure of your projects. This helps when moving into projects built by others.
```
Comment (marvingay) [inline foundations/tying_it_all_together/conclusion.md:17]: Perhaps the verbiage could be improved here, but specifically with Rails, this is critical and not worth outright deleting.

File: foundations/tying_it_all_together/conclusion.md [hunk 9]
```diff
@@ -1,43 +1,44 @@
...
-  - You will learn things "The Ruby Way", which can help you write clean code in other languages as well.
-* Elegant
-  - Brings a lot of tools to help you develop things in a way that makes your code easy to read.
-  - Often times, solutions in Ruby are much more concise than other languages.
+* Convention over configuration
+  - You don't have to make decisions on the structure of your projects.
+  - Others' Ruby code tend to be similarly structured and thus easy to navigate.
```
Comment (marvingay) [inline foundations/tying_it_all_together/conclusion.md:18]: This is true of Rails, Ruby -- not so much. An amalgamation of some sort is to be preferred here.

File: foundations/tying_it_all_together/conclusion.md (hunk 4 above)
Comment (marvingay) [inline foundations/tying_it_all_together/conclusion.md:21]: "What's a class?" - Beginner, Novice, etc.

The deleted text is to be preferred.

I like the new headings, perhaps an additional bullet or two can be added as well to write these headings home.

File: foundations/tying_it_all_together/conclusion.md [hunk 10]
```diff
@@ -1,43 +1,44 @@
...
+  - Often code can be written in a human-readable style.
+  - Easy to modify existing classes for your usecase.
 * Rapid development
-  - Rails helps you build applications very quickly, this means you can make a greater amount of interesting projects in the same time frame as others.
-* Single paradigm
-  - OOP is the Ruby way, many other languages are primarily OOP based.
+  - Rails is "batteries-included". It comes with solutions to common problems already bundled and ready to use.
+  - Rails can generate project boilerplate with it's scaffolding generators and CLI commands before you even open your editor.
+* Everything is an Object
+  - Community embraces OOP and further leads to consistency across projects.
```
Comment (marvingay) [inline foundations/tying_it_all_together/conclusion.md:26]: "What's OOP"? It's better to write that out fully here.

File: foundations/tying_it_all_together/conclusion.md [hunk 11]
```diff
@@ -1,43 +1,44 @@
...
 
 #### JavaScript/Node & Express
 * Un-opinionated
   - There is no standard way of doing things, you are going to have decide structure of projects yourself, for better or worse.
+* Large and Still Growing Community
+  - New, innovative tooling and libraries being developed all the time.
+  - Lots of new employment opportunities using JS.
```
Comment (marvingay) [inline foundations/tying_it_all_together/conclusion.md:33]: I believe the new points are redundant, but that it's worth mentioning in the "Same language" as a point that JavaScript is used everywhere on the frontend and thus an abundance of opportunity if you do not wish to be full stack.

Not a hard suggestion, but somewhat misleading when doing a compare and contrast of our tracks.

**Review by marvingay — CHANGES_REQUESTED** [review]
Everything else looks good.

File: foundations/tying_it_all_together/conclusion.md (hunk 3 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:3]: "Bootstrap" is not technical jargon. And I disagree that frameworks inherently make your code cleaner or more maintainable. You're right about "interactive *with*" vs "interactive *using*" though.

File: foundations/tying_it_all_together/conclusion.md (hunk 5 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:24]: Which word was too intimidating @I3uckwheat ? There's an explanation immediately following "batteries-included", boilerplate is not programming specific and scaffolding generators are used in a sentence explaining what they do... Give the users a *little* credit.

File: foundations/tying_it_all_together/conclusion.md (hunk 6 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:32]: I disagree that es 20xx JS is *that* much more verbose that it requires users to take it into consideration when choosing their path forward. And I hard disagree that writing JS gives you any significant insight into "how computers read your code" that you wouldn't get from any other similarly high level language.

File: foundations/tying_it_all_together/conclusion.md (hunk 7 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:9]: I intentionally removed that bit. Better to be explicit than sell some snake oil

File: foundations/tying_it_all_together/conclusion.md (hunk 5 above)
Comment (twalton83) [reply foundations/tying_it_all_together/conclusion.md:24]: Just want to hop in with a small grammatical change: "it's" should be "its".

File: foundations/tying_it_all_together/conclusion.md (hunk 8 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:17]: I agree with your next comment about swapping Ruby for Rails, and I think that should cover this one as well.

File: foundations/tying_it_all_together/conclusion.md (hunk 4 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:21]: I think the deleted text is not comparable to the supplement. Saying Ruby code is inherently more "elegant" and "concise" is not something that's always desirable. You can shrink up JS to a pretty terse format too, but better to write for future readers. We could probably keep something about Ruby having a lot of build-in methods or something though.

File: foundations/tying_it_all_together/conclusion.md (hunk 10 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:26]: OOP was mentioned in the original version, and I'm thinking if they don't know what "OOP" is they probably won't know what the expanded name means either. That said I'm open to simplifying it a bit, if you think it's necessary.

File: foundations/tying_it_all_together/conclusion.md (hunk 11 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:33]: Same language on front is a popular perk probably worth mentioning. I don't feel it's misleading to suggest JS is growing fast than Ruby is. There are plenty of surveys and studies to show that. Not to diminish the Ruby community but the point stands.

File: foundations/tying_it_all_together/conclusion.md (hunk 4 above)
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md]: Not at this level, I can concede that.

File: foundations/tying_it_all_together/conclusion.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md]: While I don't think that was super technical jargon, I made it a little easier to digest.

File: foundations/tying_it_all_together/conclusion.md (hunk 6 above)
Comment (I3uckwheat) [reply foundations/tying_it_all_together/conclusion.md:32]: I disagree though, there's a lot more setup with node when compared to Rails.

File: foundations/tying_it_all_together/conclusion.md (hunk 7 above)
Comment (I3uckwheat) [reply foundations/tying_it_all_together/conclusion.md:9]: I don't think an "engineering mindset" is snakeoil

Comment (I3uckwheat) [reply foundations/tying_it_all_together/conclusion.md:9]: @ryanford-dev But I know you have some very strong opinions on the term "engineer" so I can concede if you feel that it's really that false.

File: foundations/tying_it_all_together/conclusion.md [hunk 12]
```diff
@@ -1,43 +1,44 @@
...
 
 Arguments about learning the languages used in your area may sound convincing, but because a _good_ developer can transfer skills easily, what you learn isn't going to massively change outcomes of getting a job. Though, if you cannot shake that fear, do whatever you need, even if that includes researching your area to see what jobs are available locally (don't forget about remote jobs though). A good course is far more important than the language you learn first. 
 
 To help you make a decision, here are some perks of each technology:
 
 #### Ruby and Ruby On Rails: 
-* Opinionated 
-  - You don't have to make decisions on the structure of your projects. This helps when moving into projects built by others.
-  - You will learn things "The Ruby Way", which can help you write clean code in other languages as well.
```
Comment (I3uckwheat) [inline foundations/tying_it_all_together/conclusion.md:18]: Why did you remove this?

Comment (ryanford-dev) [reply foundations/tying_it_all_together/conclusion.md:18]: Following community standards or style guides leading to more consistent code is not exclusive to Ruby. The bullet points on the languages should be based on objective reasons, not the mantras that get repeated in their respective communities.

File: foundations/tying_it_all_together/conclusion.md (hunk 10 above)
Comment (marvingay) [reply foundations/tying_it_all_together/conclusion.md]: "OOP" just hasn't been introduced to the user (anymore) so for the sake of keeping the text formal or closer to academic, it's better to write it out.

In contrast, the learner has been introduced to what CLI is and should be aware of what CLI stands for, if that makes sense.

**Comment by I3uckwheat** [comment]
Closing due to being stale
